File: frontend/src/app/services/artist/artist.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Globals} from '../../global/globals';
import {ArtistDto} from '../../dtos/artist/artist.dto';
import {Observable} from 'rxjs';
import {ArtistEventDto} from '../../dtos/artist/artist-event.dto';
import {PageDto} from '../../dtos/page/page.dto';

@Injectable({
  providedIn: 'root'
})
export class ArtistService {

  private artistBaseUri: string = this.globals.backendUri + '/artist';

  constructor(private httpClient: HttpClient, private globals: Globals) {
  }

  /**
   * Creates an artist
   *
   * @param artistRequest is an artistDTO that should be created
   * @returns created artist with responseArtistDTO
   */
  createArtist(artistRequest: ArtistDto): Observable<ArtistDto> {
    return this.httpClient.post<ArtistDto>(this.artistBaseUri, artistRequest);
  }

  /**
   * Returns list of artists that contain a given string
   *
   * @param name substring of artists to be searched for
   * @returns list of artists that contain the substring
   */
  getArtistsByNameFilter(name: string): Observable<ArtistDto[]> {
    let queryParams = new HttpParams();
    queryParams = queryParams.set('nameFilter', name);
    return this.httpClient.get<ArtistDto[]>(this.artistBaseUri + 's', {params: queryParams});
  }

  filterArtistEvents(queryParams): Observable<PageDto<ArtistEventDto>> {
    return this.httpClient.get<PageDto<ArtistEventDto>>(this.globals.backendUri + '/artists-events',
      {params: queryParams});
  }

}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/service/artist/ArtistService.java
package at.ac.tuwien.sepm.groupphase.backend.service.artist;

import at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.artist.ArtistEventDto;
import at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.artist.ArtistRequestDto;
import at.ac.tuwien.sepm.groupphase.backend.entity.Artist;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ArtistService {
  /**
   * Creates an artist.
   *
   * @param artistRequestDto to be created
   * @return entity of created Artist
   */
  Artist create(ArtistRequestDto artistRequestDto);

  /**
   * Gets artistIds that contain a given string in any name.
   *
   * @param nameFilter string to be searched for
   * @return List of Artists that fulfill requirements
   */
  List<Artist> findArtistsByName(String nameFilter);

  /**
   * Filters artists with events by name filter.
   *
   * @param name name of the artists to be searched for
   * @param pageable defines pagination
   * @return List of Artists that fulfill requirements
   */
  Page<ArtistEventDto> filterArtistEvents(String name, Pageable pageable);
}
